### DISCUSSION TITLE
Add some features.

### DISCUSSION THREAD

**Discussion by mh292929** [Taewan-P/gpt_mobile#90] [Ideas]
Hi, please add the following features if possible: 1. Add a button to stop replying. 2. Add photo upload option. 3. Add the ability to copy code so that if we had code to program, we could copy it directly without copying the entire message.

**Comment by Taewan-P** [comment]
For no.3, You can copy parts of code by long clicking the parts of the text.

**Comment by mh292929** [comment]
And please make the api field optional. I have a url where the api field must be empty.

**Comment by Taewan-P** [comment]
How can a API field be optional?

**Comment by mh292929** [comment]
Please check github.com/xtekky/gpt4free/blob/main/docs/interference-api.md
It use open ai lib without api

**Comment by Taewan-P** [comment]
Are you talking about API keys?

- If that's the case, just put random string and see if it works
- It random string does not work, use Ollama platform with the gpt4free address since Ollama doesn't send API keys in requests

**Comment by mh292929** [comment]
I test with open ai but doesn't work. I'll test with ollama

**Comment by mh292929** [comment]
Hi. Unknown error.
If possible, send me an email or telegram address so I can send you a sample address.

**Comment by mh292929** [comment]
Because I installed this project on my server.

**Comment by Taewan-P** [comment]
If so, please send details at google-play@chungjungsoo.dev

**Comment by mh292929** [comment]
Hi, I have sent you the details. thank you

**Comment by Taewan-P** [comment]
Your server throws an error when requested a streamed response. This app only uses streamed response so that's why it throws an error.
<img width="1036" alt="image" src="https://github.com/user-attachments/assets/a9f6aa83-48db-4e40-88d9-af9f07997c33">
<img width="332" alt="image" src="https://github.com/user-attachments/assets/f087cf2c-36af-4728-bd59-2b7c52c6a5e1">

[link](https://platform.openai.com/docs/api-reference/chat/create#chat-create-stream)

**Comment by mh292929** [comment]
Does that mean there is no way to solve the problem? For example, in later versions

**Comment by Taewan-P** [comment]
It is not the app's problem. You have to fix your own server.

 I've read some references at gpt4free, and it seems like it does support streaming.

Also, there is a related issue (https://github.com/xtekky/gpt4free/issues/2278) that is very similar to your case (and is fixed), so try troubleshooting it.

**Comment by mh292929** [comment]
What address should I enter in the url for testing?

**Comment by Taewan-P** [comment]
Same as https://github.com/Taewan-P/gpt_mobile/discussions/89#discussioncomment-11075503

**Comment by mh292929** [comment]
hello Yes, the problem was with the code and I was able to solve it. thank you It will be great only if upload is added to the software.
